src/generators/AbstractRenderer.ts

```diff
@@ -14,6 +14,18 @@ export abstract class AbstractRenderer<O extends CommonGeneratorOptions = Common
     public dependencies: string[] = []
   ) {}
 
+  /**
+   * Adds a dependency while ensuring that only one dependency is preset at a time.
+   * 
+   * @param dependency complete dependency string so it can be rendered as is.
+   */
+  addUniqueDependency(dependency: string): void {
+    this.dependencies = this.dependencies || [];
+    if (!this.dependencies.includes(dependency)) {
+      this.dependencies.push(dependency);
+    }
+  }
```

Comment:
I think that we should rename it to shortest name. `addDependency` is enough :)

```suggestion
  addDependency(dependency: string): void {
    this.dependencies = this.dependencies || [];
    if (!this.dependencies.includes(dependency)) {
      this.dependencies.push(dependency);
    }
  }
```

Reply:
Also consider to write `removeDependency` function.

Reply:
I dont see a reason to add it at the moment? 🤔We can always add it later if we ever get a use-case for it.

Regarding the renaming, hmmm... You really think `addDependency` is better? If it only added to the array then it would make sense, but since it has a check before doing it it makes less sense to me 🤔

Reply:
> Regarding the renaming, hmmm... You really think addDependency is better? If it only added to the array then it would make sense, but since it has a check before doing it it makes less sense to me 🤔

![image](https://user-images.githubusercontent.com/20404945/121895304-5bef7780-cd20-11eb-8b0d-07c68c5a2638.png)

> I dont see a reason to add it at the moment? 🤔We can always add it later if we ever get a use-case for it.

I agree :)

Reply:
Fine 😂 added.